src/json_reporter.cc

```diff
@@ -232,6 +232,27 @@ void JSONReporter::PrintRunData(Run const& run) {
       << FormatKV("per_family_instance_index", run.per_family_instance_index)
       << ",\n";
   out << indent << FormatKV("run_name", run.run_name.str()) << ",\n";
+
+  // Get the name of the benchmark function
+  std::string func_name = run.run_name.str();
+  auto pos = run.run_name.str().find('/');
```

Comment:
you don't need to do any of this.  `std::string func_name = run.run_name.function_name;`

Reply:
I saw that field but realised when using BENCHMARK_CAPTURE .function_name also holds the test_case_name as well. The question is should the test_case_name be included in the reported function name or should it not.

Reply:
yes for two(ish) reasons:

1. the json "function name" should be the benchmark instance "function name" no matter how that ends up being defined by the code
2. if the function name shouldn't include the test_case_name, that's a broader discussion than the JSON reporter
  a. i think it should include the test_case_name because BENCHMARK_CAPTURE essentially defines multiple benchmarks. it's arguable that it shouldn't use "/" as the separator.

Reply:
alright will change that